bastion/usb_rules.py

```
@@ -0,0 +1,569 @@
+    def _ensure_dir(self):
+        """
+        Ensure parent directory exists with correct permissions.
+
+        Security: Rejects symlinked directories to prevent symlink attacks.
+        """
+        parent = self.db_path.parent
+
+        # SECURITY: Check if parent directory is a symlink
+        if parent.is_symlink():
+            logger.error(f"Directory {parent} is a symlink, refusing to use")
+            raise SecurityError(f"Directory {parent} is a symlink")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)